File: Managed/UnrealSharp/UnrealSharp/Interop/FRandomStreamExporter.cs
using UnrealSharp.Attributes;
using UnrealSharp.CoreUObject;

namespace UnrealSharp.Interop;

[NativeCallbacks, InternalsVisible(true)]
internal static unsafe partial class FRandomStreamExporter
{
    public static delegate* unmanaged<ref FRandomStream, void> GenerateNewSeed;
    public static delegate* unmanaged<ref FRandomStream, float> GetFraction;
    public static delegate* unmanaged<ref FRandomStream, uint> GetUnsignedInt;
    public static delegate* unmanaged<ref FRandomStream, FVector> GetUnitVector;
    public static delegate* unmanaged<ref FRandomStream, int, int, int> RandRange;
    public static delegate* unmanaged<ref FRandomStream, FVector, float, FVector> VRandCone;
    public static delegate* unmanaged<ref FRandomStream, FVector, float, float, FVector> VRandCone2;
}

File: Source/UnrealSharpCore/Export/FRandomStreamExporter.cpp
#include "FRandomStreamExporter.h"

void UFRandomStreamExporter::GenerateNewSeed(FRandomStream* RandomStream)
{
	RandomStream->GenerateNewSeed();
}

float UFRandomStreamExporter::GetFraction(FRandomStream* RandomStream)
{
	return RandomStream->GetFraction();
}

uint32 UFRandomStreamExporter::GetUnsignedInt(FRandomStream* RandomStream)
{
	return RandomStream->GetUnsignedInt();
}

FVector UFRandomStreamExporter::GetUnitVector(FRandomStream* RandomStream)
{
	return RandomStream->GetUnitVector();
}

int UFRandomStreamExporter::RandRange(FRandomStream* RandomStream, int32 Min, int32 Max)
{
	return RandomStream->RandRange(Min, Max);
}

FVector UFRandomStreamExporter::VRandCone(FRandomStream* RandomStream, FVector Dir, float ConeHalfAngleRad)
{
	return RandomStream->VRandCone(Dir, ConeHalfAngleRad);
}

FVector UFRandomStreamExporter::VRandCone2(FRandomStream* RandomStream, FVector Dir, float HorizontalConeHalfAngleRad, float VerticalConeHalfAngleRad)
{
	return RandomStream->VRandCone(Dir, HorizontalConeHalfAngleRad, VerticalConeHalfAngleRad);
}

